tokio/src/sync/mod.rs

```diff
@@ -431,6 +431,24 @@
 //!   number of permits, which tasks may request in order to enter a critical
 //!   section. Semaphores are useful for implementing limiting or bounding of
 //!   any kind.
+//!
+//! # Runtime compatibility
+//!
+//! All synchronization primitives provided in this module are runtime agnostic.
+//! You can freely move them between different instances of the Tokio runtime
+//! or even use them from non-Tokio runtimes.
+//!
+//! When used in a Tokio runtime, the synchronization primitives participate in
+//! [cooperative scheduling](crate::task#cooperative-scheduling),
+//! periodically yielding back control to the runtime to avoid starving it.
+//! In a non-Tokio runtime, the yield points where control is handed back do
+//! nothing, not changing the behavior of the synchronization primitive in any
+//! way.
```

Comment:
The current wording talks a lot about how cooperative scheduling works internally, but we should not assume that the reader knows how coop works.
```suggestion
//! When used in a Tokio runtime, the synchronization primitives participate in
//! [cooperative scheduling](crate::task#cooperative-scheduling) to avoid
//! starvation. This feature does not apply when used from non-Tokio runtimes.
```